src/mlpack/methods/ann/loss_functions/mean_absolute_error.hpp

```diff
@@ -0,0 +1,97 @@
+/**
+ * @file methods/ann/loss_functions/mean_absolute_error.hpp
+ * @author Aakash Kaushik
+ *
+ * Implementation of the Mean Absoulte Error function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LOSS_FUNCTION_MEAN_ABSOLUTE_ERROR_HPP
+#define MLPACK_METHODS_ANN_LOSS_FUNCTION_MEAN_ABSOLUTE_ERROR_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The mean absolute error perforamce function measures the network's
+ * performance according to the mean of the absolute difference between
+ * x and y divided by x.
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class MeanAbsoluteError
+{
+ public:
+  /**
+   * Create the MeanAbsoluteError object.
+   * 
+   * @param mean Reduction type. If true, it returns the mean of 
+   * the loss. Else, it returns the sum.
+   */
+  MeanAbsoluteError(const bool mean = true);
```

Comment:
Could you take a look at #2495 and re-write this to make it similar to the changes there?
Specifically, this parameter should be renamed to ```reduction``` and by default, you need to return the sum reduction instead of the mean reduction.